src/node_modules/app/machines/explore/search.options.js

```diff
@@ -0,0 +1,288 @@
+	);
+	index--;
+	if (index >= 0) {
+		ctx.userSelection.set(fieldStats[index].name);
+		ctx.selectedFieldName.set(computeFieldSelection(
+			fieldStats,
+			get(ctx.userSelection))
+		);
+	}
+	return ctx;
+}
+
+function updateCurrentField (ctx, {detail}) {
+	ctx.userSelection.set(detail);
+	ctx.selectedFieldName.set(computeFieldSelection(
+		get(ctx.fieldStats),
+		get(ctx.userSelection)
+	));
+	return ctx;
+}
+
+// Suggestions utils
+const computeSuggestionsQuery = (requestedFieldName, term) => ({
+	size: 0,
+	query: {
+		term: {
+			[requestedFieldName]: term
+		}
+	},
+	aggs: {
+		[requestedFieldName]: {
+			significant_text: {
+				field: requestedFieldName,
+				// TODO include/exclude according to ES version
+				include: stopWords,
+				min_doc_count: 1,
+				size: 20
+			}
+		}
+	}
+});
+
+// Suggestions options
+const isSuggestionsRequestIncomplete = ctx =>
+	!(Boolean(get(ctx.searchQuery)) && Boolean(get(ctx.selectedFieldName)));
+
+const suggestionsRequest = ctx => request(
+	'POST',
+	getSearchURL(get(selectedDataset)),
+	{
+		data: computeSuggestionsQuery(
+			get(ctx.selectedFieldName),
+			get(ctx.searchQuery)
+		)
+	}
+);
+
+const updateSuggestionsCache = (ctx, {data}) => {
+	ctx.suggestionsCache.update(_.setKey(get(ctx.suggestionsCacheKey), data));
+	return ctx;
+};
+
+const updateSuggestions = (ctx, {data}) => {
+	const aggs = data.aggregations[get(ctx.selectedFieldName)];
+	const buckets = (aggs && aggs.buckets || [])
+	.filter(sugg => sugg.key !== get(ctx.searchQuery));
+	const sortedBuckets = _.sort(buckets, [
+		_.sorterDesc(_.getKey('doc_count')),
+		_.getKey('key')]
+	);
+	const suggestions = sortedBuckets
+	.map(sugg => `${sugg.key} (${sugg.doc_count})`);
+	if (suggestions.length === 0) {
+		suggestions.push('-- no suggestions found --')
+	}
+	ctx.suggestions.set(suggestions);
+	return ctx;
+}
+
+const setAsWaiting = ctx => {
+	const suggestions = ["Waiting..."];
+	ctx.suggestions.set(suggestions);
+	return ctx;
+}
+
+const setSuggestionsCacheKey = ctx => {
+	const {project, source, version} = get(ctx.dataset);
+	const searchQuery = get(ctx.searchQuery);
+	const selectedFieldName = get(ctx.selectedFieldName);
+	ctx.suggestionsCacheKey.set(
+		`${project}-${source}-${version}-${searchQuery}-${selectedFieldName}`
+	);
+	return ctx;
+};
+
+const isInSuggestionsCache = ctx =>
+	get(ctx.suggestionsCacheKey) in get(ctx.suggestionsCache);
+
+const loadSuggestionsFromCache = ctx => updateSuggestions(
+	ctx,
+	{data: get(ctx.suggestionsCache)[get(ctx.suggestionsCacheKey)]}
+);
+
+export const searchOptions = {
+	actions: {
+		loadStatsFromCache: assign(loadFieldStatsFromCache),
+		loadSuggestionsFromCache: assign(loadSuggestionsFromCache),
+		resetSearchContext: assign(createBaseSearchStores),
+		selectNextSearchField: assign(selectNextSearchField),
+		selectPrevSearchField: assign(selectPrevSearchField),
+		sendFieldUpdated: send('FIELD_UPDATED'),
+		// sendQueryUpdated: send('QUERY_UPDATED'), // Leaving tie-in for last
```

Comment:
leftover?

Reply:
This and the next segment is commented as having to be filled in the PR tying search to the Explore query